### PR TITLE
Update some more dependencies and fix warnings

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

I could have gone a bit further with #3657. I was writing up the dependencies that can't move so easily and couldn't justify these ones, so I've updated them :) 

### TODO:
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

File: crates/holochain/Cargo.toml [hunk 1]
```diff
@@ -132,7 +132,7 @@ regex = "1.5"
 reqwest = "0.12"
 serial_test = "3.0"
 test-case = "3.3"
-tokio-tungstenite = "0.13"
+tokio-tungstenite = "0.21"
```
Comment (ThetaSinner) [inline crates/holochain/Cargo.toml:135]: Only used in tests, no changes required. I got errors upgrading `tunestenite` in the Holochain client but no problems here.

File: crates/holochain_types/Cargo.toml [hunk 2]
```diff
@@ -29,7 +29,7 @@ holochain_serialized_bytes = "=0.0.53"
 holochain_sqlite = { path = "../holochain_sqlite", version = "^0.3.0-beta-dev.41"}
 holochain_util = { version = "^0.3.0-beta-dev.7", path = "../holochain_util", features = ["backtrace"] }
 holochain_zome_types = { path = "../holochain_zome_types", version = "^0.3.0-beta-dev.34", features = ["full"] }
-itertools = { version = "0.10" }
+itertools = { version = "0.12" }
```
Comment (ThetaSinner) [inline crates/holochain_types/Cargo.toml:32]: Just missed this one

File: crates/kitsune_p2p/dht/Cargo.toml [hunk 3]
```diff
@@ -34,7 +34,7 @@ proptest-derive = { version = "0", optional = true }
 kitsune_p2p_dht = { path = ".", features = ["test_utils", "sqlite"]}
 
 kitsune_p2p_dht_arc = { path = "../dht_arc", features = ["test_utils"]}
-holochain_serialized_bytes = "0.0.53"
+holochain_serialized_bytes = "=0.0.53"
```
Comment (ThetaSinner) [inline crates/kitsune_p2p/dht/Cargo.toml:37]: Everywhere else in the project we've pinned this, so here I get a warning that v54 is available. Just being consistent about pinning it makes it clear that it's intentional and I don't get a warning :)

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml [hunk 4]
```diff
@@ -68,7 +68,6 @@ kitsune_p2p_bootstrap = { path = "../bootstrap", features = ["sqlite"] }
 kitsune_p2p_timestamp = { path = "../timestamp", features = ["now", "arbitrary"] }
 kitsune_p2p_types = { path = "../types", features = ["test_utils"] }
 maplit = "1"
-mockall = "0.11.3"
```
Comment (ThetaSinner) [inline crates/kitsune_p2p/kitsune_p2p/Cargo.toml:71]: Not used, always included by test_utils

**Review by neonphog — APPROVED** [review]

**Pull request merged**